Improve --underscore-namespaces implementation

tomor opened the pull request:
Improves implementaion of **--underscore-namespaces** when applied on classes that **already have namespace**.

Current implementation is changing **all** classnames that contains underscores, even if they **already use namespace**. That behaviour causes Fatal errors, e.g.:
PHP Fatal error:  Uncaught InvalidArgumentException: Class name "1" is not valid

Reproducible e.g. on doctrine2 tests directory (just an example with publicly downloadable project).

This pull request **allows to use dephpend on older projects** that combine classes **with** and **without** **namespaces** or projects without namespaces which contains libraries with namespaces in the scr directory.

mihaeu commented:
Thanks for this @tomor ! I'll merge this in today or tomorrow. Just need to tag another release first.

mihaeu commented:
Added an `isNamespaced` method to the `Dependency` interface. You can use that to get rid of the comment and the `strpos` if you like :smile: should read clearer.

tomor commented:
I've updated the implementation, replaced `strpos` with `isNamespaced`. 

The tests discovered one impact:
`_A\\b --> _B\\c` no longer equals ` _A_b --> _B_c`  after it's changed to namespaced name 
... see that I had to delete this line from the tests.

This change of behaviour was introduced already in the previous commit of this pull request, it just was not discovered by the tests.

I think that it's ok, simply because `_A` is valid namespace and I gues this test was not intended to ensure that all _ are removed from all classes. I guess it was just a way how to test that transforming classes to namespaced names works, right? 
Or not? Do you really intend to remove all _ from every name? :)
